components/brave_wallet_ui/common/hooks/send.ts

```diff
@@ -131,7 +131,18 @@ export default function useSend (
       setAddressError(getLocale('braveWalletContractAddressError'))
       return
     }
-
+    if (selectedAccount.coin === BraveWallet.CoinType.FIL) {
+      setToAddress(toAddressOrUrl)
+      const valid = isValidFilAddress(toAddressOrUrl)
+      if (!valid) {
+        setAddressWarning('')
+        setAddressError(getLocale('braveWalletNotValidFilAddress'))
+        return
+      }
+      setAddressWarning('')
+      setAddressError('')
+      return
+    }
```

Comment:
I think we need to group address checks into 3 different categories now:
- ethereum specific => ex, check for `braveWalletContractAddressError`, call `findUnstoppableDomainAddress`, `findENSAddress`, etc.
- filecoin specific => ex, check for `braveWalletNotValidFilAddress`
- generic => ex, check for `braveWalletSameAddressError`.

Feel free to create separate functions for each like this (just a suggestion):

```ts
const processEthereumAddress = React.useCallback((address: string) => {
  // ENS stuff
  //
  // Unstoppable domain stuff
  //
  // check for braveWalletContractAddressError
}, [selectedAccount])

const processFilecoinAddress = React.useCallback((address: string) => {
  // check if FIL address is valid
}, [selectedAccount])

const processAddress = React.useCallback((address: string) => {
  // check for braveWalletSameAddressError
}, [selectedAccount])

React.useEffect(() => {
  const valueToLowerCase = toAddressOrUrl.toLowerCase()

  processAddress(valueToLowerCase)
  processEthereumAddress(valueToLowerCase)
  processFilecoinAddress(valueToLowerCase)
}, [toAddressOrUrl])
```